Testscripts/Windows/PERF-STORAGE-FIO-HB.ps1

```diff
@@ -0,0 +1,316 @@
+	try {
+		$maxVMHibernateWaitMin = 30
+		$maxFIORunWaitMin = 7
+		$maxVMWakeupMin = 40
+		$maxKernelCompileMin = 90
+		$azureSyncSecond = 30
+
+		$testResult = $resultFail
+
+		Write-LogDbg "Prepare swap space for VM $($AllVMData.RoleName) in RG $($AllVMData.ResourceGroupName)."
+		# Prepare the swap space in the target VM
+		$rgName = $AllVMData.ResourceGroupName
+		$vmName = $AllVMData.RoleName
+		$location = $AllVMData.Location
+		$storageType = 'StandardSSD_LRS'
+		$dataDiskName = $vmName + '_datadisk1'
+
+		#region Generate constants.sh
+		# We need to add extra parameters to constants.sh file apart from parameter properties defined in XML.
+		# Hence, we are generating constants.sh file again in test script.
+
+		Write-LogInfo "Generating constants.sh ..."
+		$constantsFile = "$LogDir\constants.sh"
+		foreach ($TestParam in $CurrentTestData.TestParameters.param) {
+			Add-Content -Value "$TestParam" -Path $constantsFile
+			Write-LogInfo "$TestParam added to constants.sh"
+		}
+
+		Write-LogInfo "constants.sh created successfully..."
+		#endregion
+
+		#region Add a new swap disk to Azure VM
+		$diskConfig = New-AzDiskConfig -SkuName $storageType -Location $location -CreateOption Empty -DiskSizeGB 1024
+		$dataDisk1 = New-AzDisk -DiskName $dataDiskName -Disk $diskConfig -ResourceGroupName $rgName
+
+		$vm = Get-AzVM -Name $vmName -ResourceGroupName $rgName
+		Start-Sleep -s $azureSyncSecond
+		$vm = Add-AzVMDataDisk -VM $vm -Name $dataDiskName -CreateOption Attach -ManagedDiskId $dataDisk1.Id -Lun 1
+		Start-Sleep -s $azureSyncSecond
+
+		$ret_val = Update-AzVM -VM $vm -ResourceGroupName $rgName
+		Write-LogInfo "Updated the VM with a new data disk"
+		Write-LogInfo "Waiting for $azureSyncSecond seconds for configuration sync"
+		# Wait for disk sync with Azure host
+		Start-Sleep -s $azureSyncSecond
+
+		# Verify the new data disk addition
+		if ($ret_val.IsSuccessStatusCode) {
+			Write-LogInfo "Successfully add a new disk to the Resource Group, $($rgName)"
+		} else {
+			Write-LogErr "Failed to add a new disk to the Resource Group, $($rgname)"
+			throw "Failed to add a new disk"
+		}
+		#endregion
+
+		$fio1Command = @"
+source utils.sh
+SetTestStateRunning
+fio --size=1G --name=beforehb --direct=1 --ioengine=libaio --filename=fiodata --overwrite=1 --readwrite=readwrite --bs=1M --runtime=1 --iodepth=128 --numjobs=32 --runtime=300 --output-format=json+ --output=beforehb.json
+rm -f fiodata
+sync
+echo 3 > /proc/sys/vm/drop_caches
+SetTestStateCompleted
+"@
+		Set-Content "$LogDir\fio1Command.sh" $fio1Command
+
+		$fio2Command = @"
+source utils.sh
+SetTestStateRunning
+fio --size=1G --name=afterhb --direct=1 --ioengine=libaio --filename=fiodata --overwrite=1 --readwrite=readwrite --bs=1M --runtime=1 --iodepth=128 --numjobs=32 --runtime=300 --output-format=json+ --output=afterhb.json
+rm -f fiodata
+sync
+echo 3 > /proc/sys/vm/drop_caches
+SetTestStateCompleted
+"@
+		Set-Content "$LogDir\fio2Command.sh" $fio2Command
```

Comment:
you can just use one script, for change part "beforehb", "afterhb", please pass as parameter when invoke the script

Reply:
It needs to extend to network workload and stress test.  Should I use the single PS script? I think the complexity is high. I won't maintain it then. You still think we should use a single PS script for the existing tests and additional tests?